public/js/app/components/SemesterDropdown.jsx

```diff
@@ -0,0 +1,39 @@
+import React, { useState, memo } from 'react'
+import { seasonStr } from '../util/helpers'
+
+function SemesterDropdown({ chooseSemesterLabel, handleSelectedSemester, semesterList, langIndex }) {
+  const [selectedSemester, setSelectedSemester] = useState()
```

Comment:
I have n't looked into it in detail, but I guess same comments as I had on https://github.com/KTH/kurs-pm-data-admin-web/pull/371 that you fixed in that PR applies here as well.

Reply:
Could you address this as well, @amirhossein-haerian @axelbjo ?